File: interfaz2.py
import tkinter as tk
from tkinter import messagebox

class Profesor:
    def __init__(self, nombre, apellido):
        self.nombre = nombre
        self.apellido = apellido
        self.asignaturas = []

    def agregar_asignatura(self, asignatura):
        self.asignaturas.append(asignatura)

    def mostrar_info(self):
        info = f"Profesor: {self.nombre} {self.apellido}\n"
        for asignatura in self.asignaturas:
            info += f" - Asignatura: {asignatura.nombre}\n"
        return info


class Estudiante:
    def __init__(self, nombre, apellido, id_estudiante):
        self.nombre = nombre
        self.apellido = apellido
        self.id_estudiante = id_estudiante
        self.cursos = []

    def agregar_curso(self, curso):
        self.cursos.append(curso)

    def mostrar_info(self):
        info = f"Estudiante: {self.nombre} {self.apellido}, ID: {self.id_estudiante}\n"
        for curso in self.cursos:
            info += f" - Curso: {curso.nombre}\n"
        return info


class Asignatura:
    def __init__(self, nombre, profesor):
        self.nombre = nombre
        self.profesor = profesor

    def mostrar_info(self):
        return f"Asignatura: {self.nombre}, Profesor: {self.profesor.nombre} {self.profesor.apellido}\n"


class Horario:
    def __init__(self, dia, hora_inicio, hora_fin):
        self.dia = dia
        self.hora_inicio = hora_inicio
        self.hora_fin = hora_fin

    def mostrar_info(self):
        return f"Horario: {self.dia}, {self.hora_inicio} - {self.hora_fin}\n"


class Curso:
    def __init__(self, nombre, profesor, horario):
        self.nombre = nombre
        self.profesor = profesor
        self.estudiantes = []
        self.horario = horario

    def agregar_estudiante(self, estudiante):
        self.estudiantes.append(estudiante)
        estudiante.agregar_curso(self)

    def mostrar_info(self):
        info = f"Curso: {self.nombre}\n"
        info += f"Profesor: {self.profesor.nombre} {self.profesor.apellido}\n"
        info += "Estudiantes inscritos:\n"
        for estudiante in self.estudiantes:
            info += f" - {estudiante.nombre} {estudiante.apellido}\n"
        info += self.horario.mostrar_info()
        return info


class Evaluacion:
    def __init__(self, curso, estudiante, nota):
        self.curso = curso
        self.estudiante = estudiante
        self.nota = nota

    def mostrar_info(self):
        return f"Evaluación: Curso: {self.curso.nombre}, Estudiante: {self.estudiante.nombre} {self.estudiante.apellido}, Nota: {self.nota}\n"


class App(tk.Tk):
    def __init__(self):
        super().__init__()
        self.title("Gestión Académica")
        self.geometry("600x600")

        self.profesores = []
        self.estudiantes = []
        self.cursos = []
        self.evaluaciones = []

        self.create_widgets()

    def create_widgets(self):
        self.tabs = tk.Frame(self)
        self.tabs.pack()

        self.add_profesor_btn = tk.Button(self.tabs, text="Añadir Profesor", command=self.add_profesor)
        self.add_profesor_btn.pack(side="left")

        self.add_estudiante_btn = tk.Button(self.tabs, text="Añadir Estudiante", command=self.add_estudiante)
        self.add_estudiante_btn.pack(side="left")

        self.add_curso_btn = tk.Button(self.tabs, text="Añadir Curso", command=self.add_curso)
        self.add_curso_btn.pack(side="left")

        self.add_evaluacion_btn = tk.Button(self.tabs, text="Añadir Evaluación", command=self.add_evaluacion)
        self.add_evaluacion_btn.pack(side="left")

        self.show_info_btn = tk.Button(self.tabs, text="Mostrar Información", command=self.mostrar_info)
        self.show_info_btn.pack(side="left")

        self.info_text = tk.Text(self, height=20, width=70)
        self.info_text.pack()

    def add_profesor(self):
        self.new_window("Añadir Profesor", self.add_profesor_action)

    def add_profesor_action(self, nombre, apellido):
        profesor = Profesor(nombre, apellido)
        self.profesores.append(profesor)
        messagebox.showinfo("Éxito", f"Profesor {nombre} {apellido} añadido.")

    def add_estudiante(self):
        self.new_window("Añadir Estudiante", self.add_estudiante_action)

    def add_estudiante_action(self, nombre, apellido, id_estudiante):
        estudiante = Estudiante(nombre, apellido, id_estudiante)
        self.estudiantes.append(estudiante)
        messagebox.showinfo("Éxito", f"Estudiante {nombre} {apellido} con ID {id_estudiante} añadido.")

    def add_curso(self):
        self.new_window("Añadir Curso", self.add_curso_action)

    def add_curso_action(self, nombre, profesor_nombre, horario_dia, horario_inicio, horario_fin):
        profesor = next((p for p in self.profesores if p.nombre == profesor_nombre), None)
        if not profesor:
            messagebox.showerror("Error", f"Profesor {profesor_nombre} no encontrado.")
            return

        horario = Horario(horario_dia, horario_inicio, horario_fin)
        curso = Curso(nombre, profesor, horario)
        self.cursos.append(curso)
        messagebox.showinfo("Éxito", f"Curso {nombre} añadido.")

    def add_evaluacion(self):
        self.new_window("Añadir Evaluación", self.add_evaluacion_action)

    def add_evaluacion_action(self, curso_nombre, estudiante_nombre, nota):
        curso = next((c for c in self.cursos if c.nombre == curso_nombre), None)
        estudiante = next((e for e in self.estudiantes if e.nombre == estudiante_nombre), None)

        if not curso or not estudiante:
            messagebox.showerror("Error", "Curso o Estudiante no encontrado.")
            return

        evaluacion = Evaluacion(curso, estudiante, int(nota))
        self.evaluaciones.append(evaluacion)
        messagebox.showinfo("Éxito", f"Evaluación añadida: {nota} para {estudiante_nombre} en {curso_nombre}.")

    def mostrar_info(self):
        self.info_text.delete(1.0, tk.END)
        for curso in self.cursos:
            self.info_text.insert(tk.END, curso.mostrar_info())
        for evaluacion in self.evaluaciones:
            self.info_text.insert(tk.END, evaluacion.mostrar_info())

    def new_window(self, title, action):
        window = tk.Toplevel(self)
        window.title(title)
        window.geometry("400x300")

        entries = []

        def add_entry(label_text):
            frame = tk.Frame(window)
            frame.pack(pady=5)
            label = tk.Label(frame, text=label_text)
            label.pack(side="left")
            entry = tk.Entry(frame)
            entry.pack(side="left")
            entries.append(entry)

        if action == self.add_profesor_action:
            add_entry("Nombre:")
            add_entry("Apellido:")
        elif action == self.add_estudiante_action:
            add_entry("Nombre:")
            add_entry("Apellido:")
            add_entry("ID Estudiante:")
        elif action == self.add_curso_action:
            add_entry("Nombre Curso:")
            add_entry("Nombre Profesor:")
            add_entry("Día Horario:")
            add_entry("Hora Inicio:")
            add_entry("Hora Fin:")
        elif action == self.add_evaluacion_action:
            add_entry("Nombre Curso:")
            add_entry("Nombre Estudiante:")
            add_entry("Nota:")

        def submit_action():
            values = [e.get() for e in entries]
            action(*values)
            window.destroy()

        submit_btn = tk.Button(window, text="Añadir", command=submit_action)
        submit_btn.pack(pady=20)

if __name__ == "__main__":
    app = App()
    app.mainloop()

File: interfaz2.java
import javax.swing.*;
import java.awt.*;
import java.awt.event.*;
import java.util.ArrayList;
import java.util.List;

class Profesor {
    private String nombre;
    private String apellido;
    private List<Asignatura> asignaturas;

    public Profesor(String nombre, String apellido) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.asignaturas = new ArrayList<>();
    }

    public void agregarAsignatura(Asignatura asignatura) {
        asignaturas.add(asignatura);
    }

    public String mostrarInfo() {
        StringBuilder info = new StringBuilder("Profesor: " + nombre + " " + apellido + "\n");
        for (Asignatura asignatura : asignaturas) {
            info.append(" - Asignatura: ").append(asignatura.getNombre()).append("\n");
        }
        return info.toString();
    }

    public String getNombre() {
        return nombre;
    }

    public String getApellido() {
        return apellido;
    }
}

class Estudiante {
    private String nombre;
    private String apellido;
    private String idEstudiante;
    private List<Curso> cursos;

    public Estudiante(String nombre, String apellido, String idEstudiante) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.idEstudiante = idEstudiante;
        this.cursos = new ArrayList<>();
    }

    public void agregarCurso(Curso curso) {
        cursos.add(curso);
    }

    public String mostrarInfo() {
        StringBuilder info = new StringBuilder("Estudiante: " + nombre + " " + apellido + ", ID: " + idEstudiante + "\n");
        for (Curso curso : cursos) {
            info.append(" - Curso: ").append(curso.getNombre()).append("\n");
        }
        return info.toString();
    }

    public String getNombre() {
        return nombre;
    }

    public String getApellido() {
        return apellido;
    }
}

class Asignatura {
    private String nombre;
    private Profesor profesor;

    public Asignatura(String nombre, Profesor profesor) {
        this.nombre = nombre;
        this.profesor = profesor;
    }

    public String mostrarInfo() {
        return "Asignatura: " + nombre + ", Profesor: " + profesor.getNombre() + " " + profesor.getApellido() + "\n";
    }

    public String getNombre() {
        return nombre;
    }

    public Profesor getProfesor() {
        return profesor;
    }
}

class Horario {
    private String dia;
    private String horaInicio;
    private String horaFin;

    public Horario(String dia, String horaInicio, String horaFin) {
        this.dia = dia;
        this.horaInicio = horaInicio;
        this.horaFin = horaFin;
    }

    public String mostrarInfo() {
        return "Horario: " + dia + ", " + horaInicio + " - " + horaFin + "\n";
    }
}

class Curso {
    private String nombre;
    private Profesor profesor;
    private List<Estudiante> estudiantes;
    private Horario horario;

    public Curso(String nombre, Profesor profesor, Horario horario) {
        this.nombre = nombre;
        this.profesor = profesor;
        this.estudiantes = new ArrayList<>();
        this.horario = horario;
    }

    public void agregarEstudiante(Estudiante estudiante) {
        estudiantes.add(estudiante);
        estudiante.agregarCurso(this);
    }

    public String mostrarInfo() {
        StringBuilder info = new StringBuilder("Curso: " + nombre + "\n");
        info.append("Profesor: ").append(profesor.getNombre()).append(" ").append(profesor.getApellido()).append("\n");
        info.append("Estudiantes inscritos:\n");
        for (Estudiante estudiante : estudiantes) {
            info.append(" - ").append(estudiante.getNombre()).append(" ").append(estudiante.getApellido()).append("\n");
        }
        info.append(horario.mostrarInfo());
        return info.toString();
    }

    public String getNombre() {
        return nombre;
    }

    public Profesor getProfesor() {
        return profesor;
    }
}

class Evaluacion {
    private Curso curso;
    private Estudiante estudiante;
    private int nota;

    public Evaluacion(Curso curso, Estudiante estudiante, int nota) {
        this.curso = curso;
        this.estudiante = estudiante;
        this.nota = nota;
    }

    public String mostrarInfo() {
        return "Evaluación: Curso: " + curso.getNombre() + ", Estudiante: " + estudiante.getNombre() + " " + estudiante.getApellido() + ", Nota: " + nota + "\n";
    }
}

public class interfaz2 {
    private List<Profesor> profesores = new ArrayList<>();
    private List<Estudiante> estudiantes = new ArrayList<>();
    private List<Curso> cursos = new ArrayList<>();
    private List<Evaluacion> evaluaciones = new ArrayList<>();

    private JTextArea infoText;

    public static void main(String[] args) {
        new interfaz2().createGUI();
    }

    private void createGUI() {
        JFrame frame = new JFrame("Gestión Académica");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setSize(700, 600);

        JPanel panel = new JPanel();
        frame.add(panel);
        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));

        JPanel buttonPanel = new JPanel();
        panel.add(buttonPanel);
        buttonPanel.setLayout(new FlowLayout());

        JButton addProfesorButton = new JButton("Añadir Profesor");
        addProfesorButton.addActionListener(e -> addProfesor());
        buttonPanel.add(addProfesorButton);

        JButton addEstudianteButton = new JButton("Añadir Estudiante");
        addEstudianteButton.addActionListener(e -> addEstudiante());
        buttonPanel.add(addEstudianteButton);

        JButton addCursoButton = new JButton("Añadir Curso");
        addCursoButton.addActionListener(e -> addCurso());
        buttonPanel.add(addCursoButton);

        JButton addEvaluacionButton = new JButton("Añadir Evaluación");
        addEvaluacionButton.addActionListener(e -> addEvaluacion());
        buttonPanel.add(addEvaluacionButton);

        JButton showInfoButton = new JButton("Mostrar Información");
        showInfoButton.addActionListener(e -> mostrarInfo());
        buttonPanel.add(showInfoButton);

        infoText = new JTextArea(20, 60);
        infoText.setEditable(false);
        JScrollPane scrollPane = new JScrollPane(infoText);
        panel.add(scrollPane);

        frame.setVisible(true);
    }

    private void addProfesor() {
        JTextField nombreField = new JTextField(5);
        JTextField apellidoField = new JTextField(5);

        JPanel myPanel = new JPanel();
        myPanel.add(new JLabel("Nombre:"));
        myPanel.add(nombreField);
        myPanel.add(new JLabel("Apellido:"));
        myPanel.add(apellidoField);

        int result = JOptionPane.showConfirmDialog(null, myPanel,
                "Añadir Profesor", JOptionPane.OK_CANCEL_OPTION);
        if (result == JOptionPane.OK_OPTION) {
            String nombre = nombreField.getText();
            String apellido = apellidoField.getText();
            Profesor profesor = new Profesor(nombre, apellido);
            profesores.add(profesor);
            JOptionPane.showMessageDialog(null, "Profesor añadido con éxito.");
        }
    }

    private void addEstudiante() {
        JTextField nombreField = new JTextField(5);
        JTextField apellidoField = new JTextField(5);
        JTextField idField = new JTextField(5);

        JPanel myPanel = new JPanel();
        myPanel.add(new JLabel("Nombre:"));
        myPanel.add(nombreField);
        myPanel.add(new JLabel("Apellido:"));
        myPanel.add(apellidoField);
        myPanel.add(new JLabel("ID Estudiante:"));
        myPanel.add(idField);

        int result = JOptionPane.showConfirmDialog(null, myPanel,
                "Añadir Estudiante", JOptionPane.OK_CANCEL_OPTION);
        if (result == JOptionPane.OK_OPTION) {
            String nombre = nombreField.getText();
            String apellido = apellidoField.getText();
            String id = idField.getText();
            Estudiante estudiante = new Estudiante(nombre, apellido, id);
            estudiantes.add(estudiante);
            JOptionPane.showMessageDialog(null, "Estudiante añadido con éxito.");
        }
    }

    private void addCurso() {
        JTextField nombreField = new JTextField(5);
        JTextField profesorField = new JTextField(5);
        JTextField diaField = new JTextField(5);
        JTextField inicioField = new JTextField(5);
        JTextField finField = new JTextField(5);

        JPanel myPanel = new JPanel();
        myPanel.add(new JLabel("Nombre Curso:"));
        myPanel.add(nombreField);
        myPanel.add(new JLabel("Nombre Profesor:"));
        myPanel.add(profesorField);
        myPanel.add(new JLabel("Día Horario:"));
        myPanel.add(diaField);
        myPanel.add(new JLabel("Hora Inicio:"));
        myPanel.add(inicioField);
        myPanel.add(new JLabel("Hora Fin:"));
        myPanel.add(finField);

        int result = JOptionPane.showConfirmDialog(null, myPanel,
                "Añadir Curso", JOptionPane.OK_CANCEL_OPTION);
        if (result == JOptionPane.OK_OPTION) {
            String nombre = nombreField.getText();
            String profesorNombre = profesorField.getText();
            String dia = diaField.getText();
            String inicio = inicioField.getText();
            String fin = finField.getText();

            Profesor profesor = profesores.stream()
                    .filter(p -> p.getNombre().equals(profesorNombre))
                    .findFirst()
                    .orElse(null);

            if (profesor == null) {
                JOptionPane.showMessageDialog(null, "Profesor no encontrado.");
                return;
            }

            Horario horario = new Horario(dia, inicio, fin);
            Curso curso = new Curso(nombre, profesor, horario);
            cursos.add(curso);
            JOptionPane.showMessageDialog(null, "Curso añadido con éxito.");
        }
    }

    private void addEvaluacion() {
        JTextField cursoField = new JTextField(5);
        JTextField estudianteField = new JTextField(5);
        JTextField notaField = new JTextField(5);

        JPanel myPanel = new JPanel();
        myPanel.add(new JLabel("Nombre Curso:"));
        myPanel.add(cursoField);
        myPanel.add(new JLabel("Nombre Estudiante:"));
        myPanel.add(estudianteField);
        myPanel.add(new JLabel("Nota:"));
        myPanel.add(notaField);

        int result = JOptionPane.showConfirmDialog(null, myPanel,
                "Añadir Evaluación", JOptionPane.OK_CANCEL_OPTION);
        if (result == JOptionPane.OK_OPTION) {
            String cursoNombre = cursoField.getText();
            String estudianteNombre = estudianteField.getText();
            int nota = Integer.parseInt(notaField.getText());

            Curso curso = cursos.stream()
                    .filter(c -> c.getNombre().equals(cursoNombre))
                    .findFirst()
                    .orElse(null);

            Estudiante estudiante = estudiantes.stream()
                    .filter(e -> e.getNombre().equals(estudianteNombre))
                    .findFirst()
                    .orElse(null);

            if (curso == null || estudiante == null) {
                JOptionPane.showMessageDialog(null, "Curso o Estudiante no encontrado.");
                return;
            }

            Evaluacion evaluacion = new Evaluacion(curso, estudiante, nota);
            evaluaciones.add(evaluacion);
            JOptionPane.showMessageDialog(null, "Evaluación añadida con éxito.");
        }
    }

    private void mostrarInfo() {
        StringBuilder info = new StringBuilder();
        for (Curso curso : cursos) {
            info.append(curso.mostrarInfo());
        }
        for (Evaluacion evaluacion : evaluaciones) {
            info.append(evaluacion.mostrarInfo());
        }
        infoText.setText(info.toString());
    }
}
